src/avram/callbacks.cr

```diff
@@ -24,7 +24,28 @@ module Avram::Callbacks
     end
   end
 
-  # Run the given block before saving or creating
+  # Run the given method before `run` is called on an `Operation`.
+  #
+  # ```
+  # before_run :validate_inputs
+  #
+  # private def validate_inputs
+  #   validate_required data
+  # end
+  # ```
+  macro before_run(method_name)
+    def before_run
+      {% if @type.methods.map(&.name).includes?(:before_run.id) %}
+        previous_def
+      {% else %}
+        super
+      {% end %}
+
+      {{ method_name.id }}
+    end
+  end
```

Comment:
This and the identical `before_save` macro can delegate to the other macro that takes a block so that the code doesn't have to be duplicated
```suggestion
  macro before_run(method_name)
    before_run { {{ method_name.id }} }
  end
```